### PR TITLE
Utilities for writing github bot applications and gcloud event subscribers

### PR DESCRIPTION
**Pull request by rickducott**
BOT NOTES: 
resolves https://github.com/solo-io/go-utils/issues/205

### CODE REVIEW

File: gcloudutils/util.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	"google.golang.org/api/cloudbuild/v1"
+)
+
+const (
+	nameConst    = "name"
+	versionConst = "version"
+	constraint   = "constraint"
+)
+
+func HandleFailedSourceBuild(ctx context.Context, build *cloudbuild.Build) error {
```
Comment (EItanya) [inline gcloudutils/util.go]: No longer used as we don't have any more build which use the cloud `source` for the builds. Could be a good oppurtunity to trim that code

File: gcloudutils/tags.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
+package gcloudutils
```
Comment (EItanya) [inline gcloudutils/tags.go]: This file is missing yuval's latest changes involving installation ID

File: gcloudutils/clients.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package gcloudutils
+
+import (
+	"context"
+
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/cloudbuild/v1"
+)
+
+func NewCloudBuildClient(ctx context.Context) (*cloudbuild.Service, error) {
```
Comment (EItanya) [inline gcloudutils/clients.go]: This code doesn't allow for selecting any other set of credentials than the default, just want to make sure that is intended.

File: gcloudutils/clients.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package gcloudutils
```
Comment (EItanya) [inline gcloudutils/clients.go]: One of the nice things about this client set in solobot is the ability to get clients based on the project_id, would be nice to somehow port that logic here as well.

**Comment by rickducott** [comment]
Looks like there's still a bunch of TODO's in that code, so I'll give it a few days.

File: gcloudutils/tags.go (hunk 2 above)
Comment (rickducott) [reply gcloudutils/tags.go]: Cool, there are a bunch of changes missing and a bunch of TODOs -- I'll wait a bit and then update this.

**Comment by rickducott** [comment]
Using this branch to stage a bunch of new utils pulled out of solobot to help write solo-io/post-release-bot and solo-io/changelog-bot

**Pull request closed without merging**